### PR TITLE
Feature Engineering [WIP by Mat]

### PR DESCRIPTION
**Pull request by dansbecker**

### CODE REVIEW

**Comment by dansbecker** [comment]
After you've finished the initial version of everything, you should make an editing pass and use go/kaggle-learn-style-guide  for style conventions

File: notebooks/feature_engineering/raw/ex1.ipynb [hunk 1]
```diff
@@ -0,0 +1,412 @@
...
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Baseline Model\n",
+    "\n",
+    "The first thing you need to do is construct a baseline model. All new features, processing, encodings, and feature selection should improve upon this baseline model. First you need to do a bit of feature engineering before training the model itself.\n",
+    "\n",
+    "### 1) Exercise: Features from timestamps\n",
+    "From the timestamps, create features for the day, hour, minute and second. Store these as new integer columns `day`, `hour`, `minute`, and `second` and a new DataFrame `clicks`."
+   ]
```
Comment (dansbecker) [inline notebooks/feature_engineering/raw/ex1.ipynb]: It's kind of a drag to go looking up the exact syntax for this, when it feels like we're just building the baseline.

It'd be easier (while no less conceptually interesting) if you give them an example of the syntax and have them fill out the rest. Something like
clicks = pd.DataFrame({
'day': click_data.click_time.dt.day,
'hour': ____,
'minute': ____,
'second': ____
})
```

At that point it's really easy, which I think is a shortcoming. But I don't think I gained a lot by looking up the `dt` syntax.

File: learntools/feature_engineering/ex1.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    def check(self, clicks, click_data):
+        click_times = click_data['click_time']
+        solution = click_data.assign(day=click_times.dt.day.astype('uint8'),
+                                     hour=click_times.dt.hour.astype('uint8'), 
+                                     minute=click_times.dt.minute.astype('uint8'),
+                                     second=click_times.dt.second.astype('uint8'))
+
+        
+
+        assert (solution.drop('attributed_time', axis=1) ==
```
Comment (dansbecker) [inline learntools/feature_engineering/ex1.py]: I'm getting an exception `KeyError: "['attributed_time'] not found in axis"` when trying to check my code.

I wrote my code as I would expect many users to:
```clicks = pd.DataFrame()
clicks['day'] = click_data.click_time.dt.day
clicks['hour'] = click_data.click_time.dt.hour
clicks['minute'] = click_data.click_time.dt.minute
clicks['second'] = click_data.click_time.dt.second
q_1.check()```

Having trouble seeing where `attributed_time` got created in the recommended solution.

File: notebooks/feature_engineering/raw/ex1.ipynb [hunk 3]
```diff
@@ -0,0 +1,412 @@
...
+   "source": [
+    "# Run this to check your work\n",
+    "q_1.check()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 2) Exercise: Label Encoding\n",
```
Comment (dansbecker) [inline notebooks/feature_engineering/raw/ex1.ipynb]: For users who don't remember the name LabelEncoder, it'd be useful to link back to the explanation in Intermediate ML

File: learntools/feature_engineering/ex1.py (hunk 2 above)
Comment (mcleonard) [reply learntools/feature_engineering/ex1.py]: Ah, I see what happened. I want students to add the new columns to the original data which has the `attributed_time` feature. I'll fix this.

File: notebooks/feature_engineering/raw/ex1.ipynb (hunk 3 above)
Comment (mcleonard) [reply notebooks/feature_engineering/raw/ex1.ipynb]: They'll see this in the first tutorial

**Pull request merged**